### PR TITLE
mgr/dashboard: Add ErasureCodeProfile controller

### PR DESCRIPTION
**Pull request by sebastian-philipp**
This PR is part of backend implementation of the create-pool dialog. It is related to PR #20865 but not a dependency of it. Part of the Create-Pool-Dialog is a sub-dialog to manage ECPs. 

TODO: 

- [x] Add Teuthology test.
- [x] **Requires: PR #20865**
- [x] **Requires: PR #20976**
- [x] http://tracker.ceph.com/issues/23345 (local testing only)

Signed-off-by: Sebastian Wagner <sebastian.wagner@suse.com>

**I've also added a commit made by @Devp00l to fix an issue with  the pool creation dialog.**

### CODE REVIEW

File: src/pybind/mgr/dashboard_v2/controllers/erasure_code_profile.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+    def list(self):
+        ret = []
+        for name, ecp in mgr.get('osd_map').get('erasure_code_profiles', {}).items():
+            ecp['name'] = name
+            ret.append(ecp)
+        return ret
+
+    @RESTController.args_from_json
+    def create(self, name, k, m, ruleset_failure_domain=None):
```
Comment (jecluis) [inline src/pybind/mgr/dashboard_v2/controllers/erasure_code_profile.py]: what about allowing the caller to specify the `plugin` to the be used?

File: src/pybind/mgr/dashboard_v2/controllers/erasure_code_profile.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+        profile = ['k={}'.format(k), 'm={}'.format(m)]
+        if ruleset_failure_domain:
+            profile.append('crush-failure-domain={}'.format(ruleset_failure_domain))
+
+        CephService.send_command('mon', 'osd erasure-code-profile set', name=name,
```
Comment (jecluis) [inline src/pybind/mgr/dashboard_v2/controllers/erasure_code_profile.py]: how is the potential errors of this call being handled?

**Comment by rjfd** [comment]
@sebastian-philipp can you please add a bit more detail in the PR's description about what this PR implements?

File: src/pybind/mgr/dashboard_v2/controllers/erasure_code_profile.py (hunk 2 above)
Comment (sebastian-philipp) [reply src/pybind/mgr/dashboard_v2/controllers/erasure_code_profile.py]: Error handling will change as soon as the task manager form @rjfd is merged. Right now, an exception is thrown and cached by the general exception handler which will render a JSON error containing an error message.

File: src/pybind/mgr/dashboard_v2/controllers/erasure_code_profile.py (hunk 1 above)
Comment (LenzGr) [reply src/pybind/mgr/dashboard_v2/controllers/erasure_code_profile.py]: I agree, this was something that was brought up in a conversation I had today. Being able to provide the `plugin` would be nice, but then we would also need a method that returns a list of available plugins, so we don't have to maintain a hard-coded list. Not sure if that exists, though. We had similar issues with providing a selection list for compression plugins in openATTIC where we maintained a static list, but the actual build of Ceph may have been compiled with different defaults.

Comment (sebastian-philipp) [reply src/pybind/mgr/dashboard_v2/controllers/erasure_code_profile.py]: fixed

**Comment by LenzGr** [comment]
retest this please

**Comment by sebastian-philipp** [comment]
jenkins retest this please

**Comment by rjfd** [comment]
QA tests successful: http://pulpito.ceph.com/rdias-2018-04-11_08:29:22-rados:mgr-wip-rdias-testing-distro-basic-mira/

**Pull request merged**